### PR TITLE
feat: uk disclaimer banner

### PR DESCRIPTION
**Pull request by JackShort**
* we need to add a banner for the uk on the top of the app that follows the guidelines here: https://www.figma.com/file/lTKWtn5cajzUdBCSov0jiv/UK-Legal-Request?type=design&node-id=0-1&mode=design&t=sWOrFgsSlzi1SbDR-0

Screen Shot 2023-10-05 at 4 50 23 PM

### CODE REVIEW

File: src/components/NavBar/UkBanner.tsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+      <BannerTextWrapper fontWeight={500} lineHeight="24px">
+        <Trans>
+          UK disclaimer: This web application is provided as a tool for users to interact with the Uniswap Protocol on
+          their own initiative, with no endorsement or recommendation of cryptocurrency trading activities. In doing so,
+          Uniswap is not recommending that users or potential users engage in cryptoasset trading activity, and users or
+          potential users of the web application should not regard this webpage or its contents as involving any form of
+          recommendation, invitation or inducement to deal in cryptoassets.
+        </Trans>
+      </BannerTextWrapper>
+      <ReadMoreWrapper fontWeight={500} lineHeight="24px" color="accent1" onClick={openDisclaimer}>
```
Comment (tinaszheng) [inline src/components/NavBar/UkBanner.tsx:71]: if this is clickable it should either be a link or a button for a11y 👀

**Review by tinaszheng — APPROVED** [review]

**Review by mattbspector — APPROVED** [review]
This LGTM

File: src/components/NavBar/UkDisclaimerModal.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+          <ThemedText.HeadlineLarge padding="0px 8px" fontSize="24px" lineHeight="32px" fontWeight={500}>
+            <Trans>Disclaimer for UK residents</Trans>
+          </ThemedText.HeadlineLarge>
+          <ThemedText.BodyPrimary padding="8px 8px 12px" lineHeight="24px" fontWeight={500}>
+            <Trans>
+              This web application is provided as a tool for users to interact with the Uniswap Protocol on their own
+              initiative, with no endorsement or recommendation of cryptocurrency trading activities. In doing so,
+              Uniswap is not recommending that users or potential users engage in cryptoasset trading activity, and
+              users or potential users of the web application should not regard this webpage or its contents as
+              involving any form of recommendation, invitation or inducement to deal in cryptoassets.
```
Comment (cbachmeier) [inline src/components/NavBar/UkDisclaimerModal.tsx:43]: nit: can we have the shared text as a global `const` shared between the two components? this way it's easy to modify centrally

File: src/components/NavBar/UkBanner.tsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+          UK disclaimer: This web application is provided as a tool for users to interact with the Uniswap Protocol on
+          their own initiative, with no endorsement or recommendation of cryptocurrency trading activities. In doing so,
+          Uniswap is not recommending that users or potential users engage in cryptoasset trading activity, and users or
+          potential users of the web application should not regard this webpage or its contents as involving any form of
+          recommendation, invitation or inducement to deal in cryptoassets.
+        </Trans>
+      </BannerTextWrapper>
+      <ReadMoreWrapper>
+        <ThemedText.BodySecondary fontWeight={500} lineHeight="24px" color="accent1" onClick={openDisclaimer}>
+          <Trans>Read more</Trans>
```
Comment (cbachmeier) [inline src/components/NavBar/UkBanner.tsx:73]: is Read More supposed to have an underline?
Screenshot 2023-10-05 at 2 39 24 PM

File: src/components/NavBar/UkDisclaimerModal.tsx [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+export function UkDisclaimerModal() {
+  const isOpen = useModalIsOpen(ApplicationModal.UK_DISCLAIMER)
+  const closeModal = useCloseModal()
+
+  return (
+    <Modal isOpen={isOpen} onDismiss={closeModal}>
+      <Wrapper gap="md">
+        <Row justify="flex-end" padding="8px 0px 4px">
+          <CloseIcon size={24} onClick={() => closeModal()} />
```
Comment (cbachmeier) [inline src/components/NavBar/UkDisclaimerModal.tsx:31]: CloseIcon does not have the Clickable styles, only the cursor change

File: src/components/NavBar/UkDisclaimerModal.tsx [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+        </Row>
+        <Column gap="sm">
+          <ThemedText.HeadlineLarge padding="0px 8px" fontSize="24px" lineHeight="32px" fontWeight={500}>
```
Comment (cbachmeier) [inline src/components/NavBar/UkDisclaimerModal.tsx:34]: fontWeight 500 is not in the spores color refresh, use 485 for normal or 535 for bold

Comment (cbachmeier) [reply src/components/NavBar/UkDisclaimerModal.tsx:34]: also you used a themedtext here but overwrote all the styles

File: src/components/NavBar/UkBanner.tsx [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+
+export const UK_BANNER_HEIGHT = 64
+export const UK_BANNER_HEIGHT_MD = 112
+export const UK_BANNER_HEIGHT_SM = 136
+
+const BannerWrapper = styled.div`
+  position: relative;
+  display: flex;
+  background-color: ${({ theme }) => theme.surface1};
+  padding: 20px;
```
Comment (cbachmeier) [inline src/components/NavBar/UkBanner.tsx:16]: can we reduce padding on mobile? Takes up a lot of the screen rn
Screenshot 2023-10-05 at 2 43 55 PM

File: src/components/NavBar/UkBanner.tsx (hunk 3 above)
Comment (cbachmeier) [reply src/components/NavBar/UkBanner.tsx:73]: update, this only happens after you've clicked

File: src/components/NavBar/UkDisclaimerModal.tsx [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+            <Trans>Disclaimer for UK residents</Trans>
+          </ThemedText.HeadlineLarge>
+          <ThemedText.BodyPrimary padding="8px 8px 12px" lineHeight="24px" fontWeight={500}>
```
Comment (cbachmeier) [inline src/components/NavBar/UkDisclaimerModal.tsx:37]: use spore styles

File: src/components/NavBar/UkBanner.tsx [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+  flex-shrink: 0;
+  width: max-content;
+`
+
+export function UkBanner() {
+  const openDisclaimer = useOpenModal(ApplicationModal.UK_DISCLAIMER)
+
+  return (
+    <BannerWrapper>
+      <BannerTextWrapper fontWeight={500} lineHeight="24px">
```
Comment (cbachmeier) [inline src/components/NavBar/UkBanner.tsx:62]: use spore styles

File: src/components/NavBar/UkBanner.tsx [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+        <Trans>
+          UK disclaimer: This web application is provided as a tool for users to interact with the Uniswap Protocol on
+          their own initiative, with no endorsement or recommendation of cryptocurrency trading activities. In doing so,
+          Uniswap is not recommending that users or potential users engage in cryptoasset trading activity, and users or
+          potential users of the web application should not regard this webpage or its contents as involving any form of
+          recommendation, invitation or inducement to deal in cryptoassets.
+        </Trans>
+      </BannerTextWrapper>
+      <ReadMoreWrapper>
+        <ThemedText.BodySecondary fontWeight={500} lineHeight="24px" color="accent1" onClick={openDisclaimer}>
```
Comment (cbachmeier) [inline src/components/NavBar/UkBanner.tsx:72]: use spore styles

Comment (cbachmeier) [inline src/components/NavBar/UkBanner.tsx:72]: use spore styles

File: src/pages/App.tsx [hunk 10]
```diff
@@ -78,13 +89,17 @@ export default function App() {
...
   const isDarkMode = useIsDarkMode()
   const [routerPreference] = useRouterPreference()
-  const [scrolledState, setScrolledState] = useState(false)
+  const [scrollY, setScrollY] = useState(0)
+  const scrolledState = scrollY > 0
 
   const routerConfig = useRouterConfig()
 
+  const originCountry = useAppSelector((state: AppState) => state.user.originCountry)
+  const renderUkBannner = Boolean(originCountry) && originCountry === 'GB'
```
Comment (cbachmeier) [inline src/pages/App.tsx:98]: Does GB work for Scotland, Wales, and Northern Ireland?

File: src/components/NavBar/UkDisclaimerModal.tsx (hunk 5 above)
Comment (JackShort) [reply src/components/NavBar/UkDisclaimerModal.tsx:34]: 500 is the size design told me to use on the figma - and all of these styles are also based on the figma

File: src/pages/App.tsx (hunk 10 above)
Comment (JackShort) [reply src/pages/App.tsx:98]: GB is only for the UK

File: src/components/NavBar/UkBanner.tsx (hunk 6 above)
Comment (JackShort) [reply src/components/NavBar/UkBanner.tsx:16]: for some reason this is the way design has it in the figma specifically for mobile too

**Comment by JackShort** [comment]
@cbachmeier the styles are all straight from the figma

File: src/components/NavBar/UkDisclaimerModal.tsx (hunk 5 above)
Comment (cbachmeier) [reply src/components/NavBar/UkDisclaimerModal.tsx:47]: These design might be pre-spore, we shouldn't introduce uses of the old font-weights. Tagging in @infredible and @bryankuuu

File: src/pages/App.tsx (hunk 10 above)
Comment (cbachmeier) [reply src/pages/App.tsx:98]: Scotland, Wales, and Northern Ireland are part of the UK but might be under different country codes

File: src/components/NavBar/UkDisclaimerModal.tsx [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+const StyledThemeButton = styled(ThemeButton)`
+  width: 100%;
+`
+
+export function UkDisclaimerModal() {
+  const isOpen = useModalIsOpen(ApplicationModal.UK_DISCLAIMER)
+  const closeModal = useCloseModal()
+
+  return (
+    <Modal isOpen={isOpen} onDismiss={closeModal}>
```
Comment (cbachmeier) [inline src/components/NavBar/UkDisclaimerModal.tsx:41]: non-blocking nit: would be nice if onClickOutside also dismissed

File: src/components/NavBar/UkBanner.tsx [hunk 12]
```diff
@@ -0,0 +1,82 @@
...
+  flex-shrink: 0;
+  width: max-content;
+
+  :focus {
+    text-decoration: none;
+  }
+`
+
+export const bannerText = t`
+  UK disclaimer: This web application is provided as a tool for users to interact with the Uniswap Protocol on
```
Comment (cbachmeier) [inline src/components/NavBar/UkBanner.tsx:62]: non-blocking nit: you get a double up of the `UK disclaimer` on the modal
Screenshot 2023-10-06 at 9 04 40 AM

**Review by cbachmeier — APPROVED** [review]

**Review by cbachmeier — APPROVED** [review]

**Review by cbachmeier — APPROVED** [review]

**Pull request merged**